cartridges/int_affirm_pipelines/cartridge/templates/default/util/affirmpromo.isml

```diff
@@ -4,8 +4,25 @@
 <isif condition="${pdict.fpname}">
 	<isset name="promoModal" value="${affirmUtils.getPromoModalByFinProgramName(pdict.fpname)}" scope="page"/>
 </isif>
-
-<isif condition="${pdict.context == 'cart' && affirmData.getCartPromoMessageStatus()}">
+<isscript>
+	var plis = pdict.Basket.getAllProductLineItems();
+	var inStockCart = true;
+	for (var i = 0; i < plis.length; i++) {
+		var pli = plis[i]
+		var isInStock = pli.product.availabilityModel.inStock
+		if (isInStock) { continue; } 
+		else {
+			inStockCart = false;
+			break;
+		}
+	}
```

Comment:
What happens if there is one item in stock and one item out of stock?

Reply:
Affirm will be suppressed for any cart that contains an out-of-stock (or not in stock) item. It will be the same for the checkout page.

Reply:
LGTM